maven-model-builder/src/main/java/org/apache/maven/utils/Os.java

```diff
@@ -0,0 +1,383 @@
+
+    /**
+     * OS family that can be tested for. {@value}
+     */
+    public static final String FAMILY_NETWARE = "netware";
+
+    /**
+     * OS family that can be tested for. {@value}
+     */
+    public static final String FAMILY_DOS = "dos";
+
+    /**
+     * OS family that can be tested for. {@value}
+     */
+    public static final String FAMILY_MAC = "mac";
+
+    /**
+     * OS family that can be tested for. {@value}
+     */
+    public static final String FAMILY_TANDEM = "tandem";
+
+    /**
+     * OS family that can be tested for. {@value}
+     */
+    public static final String FAMILY_UNIX = "unix";
+
+    /**
+     * OS family that can be tested for. {@value}
+     */
+    public static final String FAMILY_OPENVMS = "openvms";
+
+    /**
+     * OS family that can be tested for. {@value}
+     */
+    public static final String FAMILY_ZOS = "z/os";
+
+    /**
+     * OS family that can be tested for. {@value}
+     */
+    public static final String FAMILY_OS400 = "os/400";
+
+    /**
+     * OpenJDK is reported to call MacOS X "Darwin"
+     *
+     * @see <a href="https://issues.apache.org/bugzilla/show_bug.cgi?id=44889">bugzilla issue</a>
+     * @see <a href="https://issues.apache.org/jira/browse/HADOOP-3318">HADOOP-3318</a>
+     */
+    private static final String DARWIN = "darwin";
+
+    /**
+     * The set of valid families. This methods initializes the set until
+     * VALID_FAMILIES constant is set.
+     * @return The set of families.
+     */
+    public static Set<String> getValidFamilies() {
+        if (VALID_FAMILIES != null) {
+            return VALID_FAMILIES;
+        }
+
+        Set<String> valid = new HashSet<>();
+        valid.add(FAMILY_DOS);
+        valid.add(FAMILY_MAC);
+        valid.add(FAMILY_NETWARE);
+        valid.add(FAMILY_NT);
+        valid.add(FAMILY_OPENVMS);
+        valid.add(FAMILY_OS2);
+        valid.add(FAMILY_OS400);
+        valid.add(FAMILY_TANDEM);
+        valid.add(FAMILY_UNIX);
+        valid.add(FAMILY_WIN9X);
+        valid.add(FAMILY_WINDOWS);
+        valid.add(FAMILY_ZOS);
+
+        return Collections.unmodifiableSet(valid);
+    }
+
+    /**
+     * Default constructor
+     */
+    public Os() {
+        // default
+    }
+
+    /**
+     * Constructor that sets the family attribute
+     *
+     * @param family a String value
+     */
+    public Os(String family) {
+        setFamily(family);
+    }
+
+    /**
+     * Sets the desired OS family type
+     *
+     * @param f The OS family type desired<br />
+     *          Possible values:<br />
+     *          <ul>
+     *          <li>dos</li>
+     *          <li>mac</li>
+     *          <li>netware</li>
+     *          <li>os/2</li>
+     *          <li>tandem</li>
+     *          <li>unix</li>
+     *          <li>windows</li>
+     *          <li>win9x</li>
+     *          <li>z/os</li>
+     *          <li>os/400</li>
+     *          </ul>
+     */
+    private void setFamily(String f) {
```

Comment:
Noticing a bug now. Any family can be set, valid or not. This class really isn't OK. Move to a separate PR where the changes that involve this can get more attention.